### DISCUSSION TITLE
How do I make the progress bar move in the new onboarding workflow?

### DISCUSSION THREAD

**Discussion by mm2270** [IBM/mac-ibm-notifications#105] [Q&A] [answered]
Hi all. I have a general question on something. I'm intrigued by IBM Notifier's newer Onboarding workflow, and how it can have all these additional payloads and elements. But I'm having trouble understanding exactly how some of them work.

For example, adding a general progress bar at the bottom is great, but I can't figure out how to make the progress bar actually "progress" as a user clicks through the pages. If for example, I have 5 steps for someone to click through, I'd want the progress bar to move 20% after clicking each Continue button until it reaches 100%, so they know they've completed the steps.
I can see how to add the progress bar at a percent, or at 0% from the example JSON file, but I don't understand how to make it move along using that method. Do I somehow need to update a command file to do this? Using a JSON file for setting up the Onboarding is not the same as a regular script and utilizing the progress bar where we can have a named pipe hold the percentage.

Can anyone suggest a way to do this that I'm overlooking?

**Comment by SMartorelli** [comment, marked as answer]
Hi @mm2270, the progress bar on the onboarding UI just work like the [progress bar on the pop-up UI](https://github.com/IBM/mac-ibm-notifications/wiki/Pop-up-UI-Accessory-Views) so you need to update it from the script using its payload (Ex. `/percent 30`). If you would like to have an automatic progress bar that just follow the completion of the onboarding pages please feel free to make a [feature request](https://github.com/IBM/mac-ibm-notifications/issues/new?assignees=&labels=&template=feature_request.md&title=%5BFeature+Request%5D) and the community will work on it in one of the next iterations.
I'll work on updating the onboarding UI documentation in order to make the progress bar behaviour more clear after the initial setup.

**Reply by mm2270** [reply]
Thanks @SMartorelli  that's sort of what I thought, except I'm still not understanding how that works in the onboarding when using a JSON file. I tried adding something like the following to a "page" section in the JSON file

`"progressBarPayload": "/percent 40 /top_message Something something..."`

but when that page gets loaded in the flow, it doesn't move the progress bar. What is the correct syntax to use to do this? I must be missing something.

**Reply by SMartorelli** [reply]
No you can't update the progress bar based on the page so from the JSON you can just set the initial state of the item. After you triggered IBM Notifier with the json loaded from the file then you should write to the process the progress bar updates.

**Reply by mm2270** [reply]
Hmm, ok thanks. But this leads me to another question, which is, how would I know when the user has clicked Continue or the button to move on to the next page, in order to update the progress bar? Unless I'm missing something, there doesn't seem to be a way to know this action has happened since it's user driven. Unlike, say, an installation or some other command line driven process, where I can monitor the progress and update the named pipe to update the progress bar.

**Reply by SMartorelli** [reply]
It's actually not possible to know it as of today implementation of the onboarding UI. The progress bar has been implemented to be able to show a background process (done by the script) while the user navigate some useful pages. That's why I suggested you to open a FR if you think that having a workflow where the progress bar is linked to the progression of the pages would be useful. :grin:

**Reply by mm2270** [reply]
Ok then, thank you. I'll open up a feature request for that then.
